File: product/data/src/main/java/ee/ctob/data/Participant.java
package ee.ctob.data;

import ee.ctob.data.enums.Gender;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "participants", schema = "conference")
public class Participant {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer id;
    LocalDateTime created;
    @Column(name = "validation_uuid", unique = true)
    UUID validationUUID;
    @Column(name = "participant_uuid", unique = true)
    UUID participantUUID;
    String firstName;
    String lastName;
    String email;
    @Enumerated(EnumType.STRING)
    Gender gender;
    LocalDate dateOfBirth;
    String feedback;
}

File: product/data/src/main/kotlin/ee/ctob/data/kotlin/Participant.kt
package ee.ctob.data.kotlin

import ee.ctob.data.enums.Gender
import java.time.LocalDateTime
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "participants", schema = "conference")
data class Participant(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int? = null,
    val created: LocalDateTime,
    @Column(name = "validation_uuid", unique = true)
    val validationUUID: UUID,
    @Column(name = "participant_uuid", unique = true)
    val participantUUID: UUID,
    val firstName: String,
    val lastName: String,
    val email: String,
    @Enumerated(EnumType.STRING)
    val gender: Gender,
    val dateOfBirth: LocalDateTime,
    val feedback: String? = null
)